### DISCUSSION TITLE
Organizing and Running Cypress Tests for Multiple Software Versions with Different Functionality

### DISCUSSION THREAD

**Discussion by mayssacherif** [cypress-io/cypress#28718] [Questions and Help]
Hello everyOne,

I am working on a project where we have multiple versions of our software, each with its own set of functionalities. I'm using Cypress for end-to-end testing, and I want to organize my test scenarios in a way that allows me to run tests for each software version separately.

Current Project Structure:

Software Version 1
Test Scenarios A, B, C
Software Version 2
Test Scenarios X, Y, Z
Question:
How can I structure my Cypress test suite to group tests for each software version individually and indicate to Cypress which series of tests should be executed for each version?

**Comment by portyu9** [comment]
Hi @mayssacherif 

A simple supported approach is to organize the specs by software version and use Cypress's `--spec` option to choose the group to run.

For example:

```text
cypress/
  e2e/
    version-1/
      scenario-a.cy.ts
      scenario-b.cy.ts
      scenario-c.cy.ts

    version-2/
      scenario-x.cy.ts
      scenario-y.cy.ts
      scenario-z.cy.ts
```

Keep the normal broad pattern in `cypress.config.js`:

```js
const { defineConfig } = require('cypress')

module.exports = defineConfig({
  e2e: {
    specPattern: 'cypress/e2e/**/*.cy.{js,jsx,ts,tsx}',
  },
})
```

Then add package scripts for each version:

```json
{
  "scripts": {
    "cy:v1": "cypress run --spec \"cypress/e2e/version-1/**/*.cy.{js,jsx,ts,tsx}\"",
    "cy:v2": "cypress run --spec \"cypress/e2e/version-2/**/*.cy.{js,jsx,ts,tsx}\""
  }
}
```

Run:

```bash
npm run cy:v1
```

or:

```bash
npm run cy:v2
```

`specPattern` defines which files are Cypress specs in general, while `--spec` narrows that set for a particular run.

If some scenarios are valid for both versions, I would keep reusable commands/helpers shared and avoid duplicating their implementation. If entire spec files are common, you can also keep a `common/` directory and include both glob patterns in the run.

This is usually cleaner than loading every test and putting version checks/skips throughout the individual test cases.

Please mark this answer as accepted if it helped, thank you.
